subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c

```diff
@@ -531,37 +643,141 @@ static int isr_rx(struct lll_sync *lll, uint8_t node_type, uint8_t *crc_ok)
 	}
 #endif /* CONFIG_BT_CTLR_DF_SAMPLE_CTE_FOR_PDU_WITH_BAD_CRC */
 
-isr_rx_done:
 	return err;
 }
 
+static void isr_rx_adv_sync_estab(void *param)
+{
+	enum sync_status sync_ok;
+	struct lll_sync *lll;
+	uint8_t rssi_ready;
+	uint8_t trx_done;
+	uint8_t crc_ok;
+	int err;
+
+	lll = param;
+
+	/* Read radio status and events */
+	trx_done = radio_is_done();
+	if (trx_done) {
+		crc_ok = radio_crc_is_valid();
+		rssi_ready = radio_rssi_is_ready();
+		sync_ok = sync_filtrate_by_cte_type(lll->cte_type, lll->filter_policy);
+		++trx_cnt;
+	} else {
+		crc_ok = rssi_ready = 0U;
+		/* Initiated as allowed, crc_ok takes precended during handling of PDU
+		 * reception in the situation.
+		 */
+		sync_ok = SYNC_STAT_ALLOWED;
+	}
+
+	/* Clear radio rx status and events */
+	lll_isr_rx_status_reset();
+
+	/* No Rx */
+	if (!trx_done) {
+		/* TODO: Combine the early exit with above if-then-else block
+		 */
+		goto isr_rx_done;
+	}
+
+	/* Save radio ready and address capture timestamp for later use for
+	 * drift compensation.
+	 */
+	radio_tmr_aa_save(radio_tmr_aa_get());
+	radio_tmr_ready_save(radio_tmr_ready_get());
+
+	/* Handle regular PDU reception if CTE type is acceptable */
+	if (sync_ok == SYNC_STAT_ALLOWED) {
+		err = isr_rx(lll, NODE_RX_TYPE_SYNC, crc_ok, rssi_ready, SYNC_STAT_ALLOWED);
+		if (err == -EBUSY) {
+			return;
+		}
```

Comment:
What if there's another error?

Reply:
There is no possibility for other error code except `0` and `-EBUSY`. I've added a header before isr_rx function with information about possible return values.

FYI, zero is returned in case there is no chained PDU or chained PDU is spaced long enough to schedule its reception by ULL. `-EBUSY` means there is a chained PDU but too close for ULL scheduling, so it is done directly from LLL.

Reply:
> There is no possibility for other error code except 0 and -EBUSY

Isn't that a dangerous thing to assume, in case someone else updates the function at some point? I'm thinking that it would be "safer" to add an ASSERT to ensure that any other return values are "handled" or at least considered.

Reply:
Well, I'm big fan of defensive programming, nevertheless this is an ISR and I here every instruction counts. I'd like to have an opinion from @cvinayak about it.

Reply:
Aren't ASSERT compiled out for production builds, and thus safe to add? 

In any ways, if the function can only return 0 and -EBUSY, I guess the compiler would be able to optimize it anyhow, right?

Reply:
> Aren't ASSERT compiled out for production builds, and thus safe to add?

In theory the LL_ASSERTs should be removed in release builds. 

> In any ways, if the function can only return 0 and -EBUSY, I guess the compiler would be able to optimize it anyhow, right?

This is just ques what compiler may or may not do.

I have a vague memory that I've been asked to remove LL_ASSERT from code. Because of that I asked @cvinayak about his point of view about it.

Reply:
> anyone adding new return value in the future will definitely change the code to handle it.
In my opinion that's a optimistic expectation, but I suppose we'll have some CI tests that will catch it. 

I won't be blocking the PR based on this, but I do think it's a optimistic way of handling it.

Reply:
If @Thalley does not push to use assert here I'll go with @cvinayak.

Just to be clear, I'm fan of defensive programming. If I were main contributor to the code I'll go with @Thalley suggestion.